cms/templates/partials/card_details_top.html

```diff
@@ -19,17 +19,12 @@ <h3 class="title">
     </a>
   </h3>
   <ul>
-    {% if courseware_page.is_external_page and courseware_page.price %}
-    <li>
-      <strong>Price:</strong>
-      ${{ courseware_page.price|floatformat:"0"|intcomma }}
-    </li>
-    {% elif courseware_page.product.current_price %}
+    {% if courseware_page.product.current_price %}
     <li>
       <strong>Price:</strong>
       ${{ courseware_page.product.current_price|floatformat:"0"|intcomma }}
     </li>
-    {% elif courseware_page.is_course_page and courseware_page.product.first_unexpired_run.current_price %}
+    {% elif courseware_page.is_internal_or_external_course_page and courseware_page.product.first_unexpired_run.current_price %}
     <li>
       <strong>Price:</strong>
       ${{ courseware_page.product.first_unexpired_run.current_price|floatformat:"0"|intcomma }}
```

Comment:
Would creating a current price property in the Course model, using the same logic be more helpful?